var/spack/repos/builtin/packages/ninja-fortran/package.py

```diff
@@ -10,20 +10,23 @@ class NinjaFortran(Package):
     """A Fortran capable fork of ninja."""
 
     homepage = "https://github.com/Kitware/ninja"
-    url      = "https://github.com/Kitware/ninja/archive/v1.7.2.gaad58.kitware.dyndep-1.tar.gz"
+    url      = "https://github.com/Kitware/ninja/archive/v1.9.0.g99df1.kitware.dyndep-1.jobserver-1.tar.gz"
 
     # Each version is a fork off of a specific commit of ninja
     # Hashes don't sort properly, so manually set newest version
-    version('1.7.2.gaad58', 'eb51b042b9dbaf8ecd79a6fb24de1320', preferred=True)
-    version('1.7.2.gcc0ea', '3982f508c415c0abaca34cb5e92e711a')
-    version('1.7.1.g7ca7f', '187a8d15c1e20e5e9b00c5c3f227ca8a')
+    version('1.9.0.g99df1', 'f0892b1c8fd2984e8f47e9e1fcb9ce32', preferred=True)
+    version('1.9.0.g5b44b', 'f7c2e718801c1fd097a728530559e0d4')
+    version('1.9.0.gad558', 'cb93ffb5871225e2b813eb9c34b3096d')
+    version('1.8.2.g81279', '0e29d0c441dcbd9b9ee9291c3a8dfbdd')
+    version('1.8.2.g3bbbe', 'de6257118f2e3ac7fa1abca1e7c70afa')
+    version('1.8.2.g972a7', '8ace90ad0c5657022d10ba063783a652')
 
     depends_on('python', type=('build', 'run'))
 
     phases = ['configure', 'install']
 
     def url_for_version(self, version):
-        url = 'https://github.com/Kitware/ninja/archive/v{0}.kitware.dyndep-1.tar.gz'
```

Comment:
Can you modify `url_for_version` so that is possible to keep the older versions?

Reply:
I've proposed a "solution" as a modification to the branch.  I'm not sure if this is the best/right solution.  The hash strings at the end of the versions cause issues as far as sorting, e.g., a `spack info ninja-fortran` will not put the versions in the correct order (1.8.2 has multiple releases with different hashes).

I'm open to suggestions of a better approach.

Reply:
>  The hash strings at the end of the versions cause issues as far as sorting, e.g., a spack info ninja-fortran will not put the versions in the correct order (1.8.2 has multiple releases with different hashes).

how about we add another number to make sure versions are sorted correctly, i.e. `1.7.2.x.hash` ? 
those hashes don't really help to understand which is more recent and which is older anyway. But I also think it's good to keep them so that users know which upstream version is it exactly.